### PR TITLE
Define Container for Codespaces

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
This allows GitHub users to start a codespace for working on Loki. It is based on the Loki build image and hopefully makes it easier to contribute.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "image": "grafana/loki-build-image:0.27.1",
+  "containerEnv": {
+    "BUILD_IN_CONTAINER": "false"
+  },
+  "customizations": {
+		"vscode": {
+			"settings": {},
+			"extensions": ["golang.go"],
```
Comment (jeschkies) [inline .devcontainer/devcontainer.json:9]: Should we add the [Grafana Jsonnet](https://github.com/grafana/vscode-jsonnet) extension?

**Review by jeschkies — COMMENTED** [review]
The image brings everything except for gopls. This would have to be added to the build image.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "image": "grafana/loki-build-image:0.27.1",
```
Comment (MichelHollands) [inline .devcontainer/devcontainer.json:2]: There is a 0.28.1 now

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (MichelHollands) [reply .devcontainer/devcontainer.json:9]: If it works, why not?

**Review by MichelHollands — APPROVED** [review]
LGTM

**Pull request merged**